src/engine/loopingcontrol.h

```diff
@@ -32,14 +32,14 @@ class LoopingControl : public EngineControl {
     // process() updates the internal state of the LoopingControl to reflect the
     // correct current sample. If a loop should be taken LoopingControl returns
     // the sample that should be seeked to. Otherwise it returns currentSample.
-    double process(const double dRate,
+    virtual double process(const double dRate,
```

Comment:
It's definitely poor testing practice to use an object that you're trying to mock away, and by having the mock inherit from the implementation it's mocking, that's what we're doing.  Then the test is sort of worthless because instead of a test saying "this is how loopingcontrol *should* behave", it's saying "this is how loopingcontrol *does* behave".  And then we get into a circular series of assumptions.

If you look how it's been done elsewhere in Mixxx, the pattern is to create an interface class and use that.  So you could do "class LoopingControlInterface : public EngineControl" and then "class LoopingControl : public LoopingControlInterface" and "class MockLoopingControl : public LoopingControlInterface."  You don't get the nice eclipse function preview, but that's actually correct because we want that to be mockable.  But, it preserves the concept that the class is looping-specific and not just any old enginecontrol.